src/Meta/class-post-builder.php

```diff
@@ -0,0 +1,571 @@
+					}
+					if ( ! empty( $post_author ) ) {
+						$coauthors[] = $post_author;
+					}
+				}
+				// The empty else case is because if we force guest authors, we don't ever care what value wp_posts.post_author has.
+			}
+		}
+		return $coauthors;
+	}
+
+	/**
+	 * Determines author name from display name, falling back to firstname
+	 * lastname, then nickname and finally the nicename.
+	 *
+	 * @since 3.3.0 Moved to class-metadata
+	 *
+	 * @param ?WP_User $author The author of the post.
+	 * @return string
+	 */
+	private function get_author_name( ?WP_User $author ): string {
+		// Gracefully handle situation where no author is available.
+		if ( null === $author ) {
+			return '';
+		}
+
+		if ( ! empty( $author->display_name ) ) {
+			return $author->display_name;
+		}
+
+		$author_name = $author->user_firstname . ' ' . $author->user_lastname;
+		if ( ' ' !== $author_name ) {
+			return $author_name;
+		}
+
+		if ( ! empty( $author->nickname ) ) {
+			return $author->nickname;
+		}
+
+		if ( ! empty( $author->user_nicename ) ) {
+			return $author->user_nicename;
+		}
+
+		return '';
+	}
+
+	/**
+	 * Returns the tags associated with this page or post.
+	 *
+	 * @since 3.3.0 Moved to class-metadata
+	 *
+	 * @param int $post_id   The id of the post you're trying to get tags for.
+	 * @return array<string> The tags of the post represented by the post id.
+	 */
+	private function get_tags( int $post_id ): array {
+		$tags      = array();
+		$post_tags = wp_get_post_tags( $post_id );
+		if ( ! is_wp_error( $post_tags ) ) {
+			foreach ( $post_tags as $wp_tag ) {
+				$tags[] = $wp_tag->name;
+			}
+		}
+		return $tags;
+	}
+
+	/**
+	 * Returns an array of all the child categories for the current post.
+	 *
+	 * @since 3.3.0 Moved to class-metadata
+	 *
+	 * @param int    $post_id The id of the post you're trying to get categories for.
+	 * @param string $delimiter What character will delimit the categories.
+	 * @return array<string> All the child categories of the current post.
+	 */
+	private function get_categories( int $post_id, string $delimiter = '/' ): array {
+		$tags = array();
+		foreach ( get_the_category( $post_id ) as $category ) {
+			$hierarchy = get_category_parents( $category->term_id, false, $delimiter );
+			if ( ! is_wp_error( $hierarchy ) ) {
+				$tags[] = rtrim( $hierarchy, '/' );
+			}
+		}
+		// take last element in the hierarchy, a string representing the full parent->child tree,
+		// and split it into individual category names.
+		$last_tag = end( $tags );
+		if ( false !== $last_tag ) {
+			$tags = explode( '/', $last_tag );
+		}
+
+		// Remove default category name from tags if needed.
+		$default_category_name = get_cat_name( get_option( 'default_category' ) );
+		return array_diff( $tags, array( $default_category_name ) );
+	}
+
+	/**
+	 * Gets all term values from custom taxonomies.
+	 *
+	 * @since 3.3.0 Moved to class-metadata
+	 *
+	 * @param WP_Post $post_obj The post object.
+	 * @return array<string>
```

Comment:
```suggestion
	 * Gets all term names from all custom taxonomies assigned to a post.
	 *
	 * @since 3.3.0 Moved to class-metadata
	 * @since 3.4.0 Moved to class-post-builder
	 *
	 * @param WP_Post $post_obj The post object to find the terms for.
	 * @return array<string> Term names.
```